src/racing/actors.c

```diff
@@ -2517,121 +1426,121 @@ s16 addActorToEmptySlot(Vec3f pos, Vec3s rot, Vec3f velocity, s16 actorType) {
-    if (arg3 < 0.1f) {
-        return 0;
+    dist = (xDist * xDist) + (yDist * yDist) + (zDist * zDist);
+    if (dist < 0.1f) {
+        return FALSE;
     }
-    if ((arg2 * arg2) < arg3) {
-        return 0;
+    if ((minDist * minDist) < dist) {
+        return FALSE;
     }
-    temp_f0_3 = sqrtf(arg3);
-    sp28 = temp_f0_3 - arg2;
-    temp_f16 = player->velocity[0];
-    temp_f18 = player->velocity[2];
+    sqrtDist = sqrtf(dist);
+    sp28 = sqrtDist - minDist;
+    xVelocity = player->velocity[0];
+    zVelocity = player->velocity[2];
     if (player->effects & STAR_EFFECT) {
-        return 1;
+        return TRUE;
     }
-    if (temp_f0_3 < 0.1f) {
-        temp_f0_4 = sqrtf((temp_f16 * temp_f16) + (temp_f18 * temp_f18));
+    if (sqrtDist < 0.1f) {
+        temp_f0_4 = sqrtf((xVelocity * xVelocity) + (zVelocity * zVelocity));
         if (temp_f0_4 < 0.5f) {
             temp_f0_4 = 0.5f;
         }
         player->velocity[0] = 0;
         player->velocity[2] = 0;
-        player->pos[0] += (temp_f16 / temp_f0_4) * arg2;
-        player->pos[2] += (temp_f18 / temp_f0_4) * arg2;
+        player->pos[0] += (xVelocity / temp_f0_4) * minDist;
+        player->pos[2] += (zVelocity / temp_f0_4) * minDist;
     } else {
         player->effects |= 0x8000;
-        temp_f20 /= temp_f0_3;
-        temp_f14 /= temp_f0_3;
-        temp_f0_5 = sqrtf((temp_f16 * temp_f16) + (temp_f18 * temp_f18));
+        xDist /= sqrtDist;
+        zDist /= sqrtDist;
+        temp_f0_5 = sqrtf((xVelocity * xVelocity) + (zVelocity * zVelocity));
         if (temp_f0_5 < 0.25f) {
             temp_f0_6 = 1.2f;
-            player->pos[0] = pos[0] - (temp_f20 * arg2 * temp_f0_6);
-            player->pos[2] = pos[2] - (temp_f14 * arg2 * temp_f0_6);
+            player->pos[0] = pos[0] - (xDist * minDist * temp_f0_6);
+            player->pos[2] = pos[2] - (zDist * minDist * temp_f0_6);
             player->velocity[0] = 0.0f;
             player->velocity[2] = 0.0f;
-            return 1;
+            return TRUE;
         }
-        temp_f2_2 = ((temp_f20 * temp_f16) + (temp_f14 * temp_f18)) / temp_f0_5;
+        temp_f2_2 = ((xDist * xVelocity) + (zDist * zVelocity)) / temp_f0_5;
         temp_f2_2 = temp_f0_5 * temp_f2_2 * arg4 * 1.3f;
-        player->velocity[0] -= temp_f20 * temp_f2_2;
-        player->velocity[2] -= temp_f14 * temp_f2_2;
-        player->pos[0] += temp_f20 * sp28 * 0.5f;
-        player->pos[2] += temp_f14 * sp28 * 0.5f;
+        player->velocity[0] -= xDist * temp_f2_2;
+        player->velocity[2] -= zDist * temp_f2_2;
+        player->pos[0] += xDist * sp28 * 0.5f;
+        player->pos[2] += zDist * sp28 * 0.5f;
     }
-    return 1;
+    return TRUE;
 }
 
-s32 func_8029F1F8(Player *player, struct Actor *marioRacewaySign) {
-    if (func_8029EEB8(player, marioRacewaySign->pos, 7.0f, 200.0f, 0.8f) == 1) {
+bool collide_mario_sign(Player *player, struct Actor *marioRacewaySign) {
```

Comment:
mario_sign_collision

Reply:
I prefer put the action word in first so collision_mario_sign ?

Reply:
Sure